### PR TITLE
Serialization of ring_size in ShareTensor, ReplicatedSharedTensor

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Ring size is serialized for ShareTensor and ReplicatedSharedTensor, as we work with various share types(arithmetic,boolean),correct ring_size if required at the parties end.

Lowered the version of Pillow due to https://github.com/python-pillow/Pillow/issues/5571

## How has this been tested?
Will be added in SyMPC after merge.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)

### CODE REVIEW

**Review by kamathhrishi — APPROVED** [review]
Looks good to me.

**Review by gmuraru — APPROVED** [review]
LGTM!

File: packages/syft/setup.cfg [hunk 1]
```diff
@@ -122,6 +122,7 @@ dev =
     syft-proto
     typing-extensions
     sqlitedict
+    pillow==8.2.0
```
Comment (gmuraru) [inline packages/syft/setup.cfg:125]: @tudorcebere could you look over this file? if those changes seem ok?

**Review by gmuraru — APPROVED** [review]
LGTM

**Comment by rasswanth-s** [comment]
Reverted Pillow version lowering as addressed here : https://github.com/OpenMined/PySyft/pull/5755

**Pull request merged**